packages/data-stores/src/plans/hooks/use-pricing-meta-for-grid-plans.ts

```diff
@@ -229,8 +234,9 @@ const usePricingMetaForGridPlans = ( {
 					// TODO clk: the condition on `.pricing` here needs investigation. There should be a pricing object for all returned API plans.
 					billingPeriod: plans.data?.[ planSlug ]?.pricing?.billPeriod,
 					// TODO clk: the condition on `.pricing` here needs investigation. There should be a pricing object for all returned API plans.
-					currencyCode: plans.data?.[ planSlug ]?.pricing?.currencyCode,
-					purchaseCurrencyCode: purchasedPlan?.currencyCode,
+					currencyCode:
+						planPrices?.[ planSlug ]?.currencyCode ??
+						plans.data?.[ planSlug ]?.pricing?.currencyCode,
```

Comment:
Considering the other comment, this would become just `currencyCode: planPrices?.[ planSlug ]?.currencyCode`.

Reply:
I updated with your suggestions in this [here](https://github.com/Automattic/wp-calypso/pull/91541/commits/52f039e920332dafbf8f9c9e7a8453215535901b)